server/src/database.rs

```diff
@@ -214,6 +212,37 @@ impl BlockDatabase {
         Ok(())
     }
 
+    /// Removes existing transaction from store
+    pub fn remove(&mut self, hash: &H256) -> Result<Option<Transaction>> {
+        fn search_transaction(hash: &H256, file: &fs::File, len: u64) -> Option<(Transaction, CursorState)> {
+            let mut it = TransactionsIterator::new(file, len, IteratorMode::ReadOnly);
+            while let Some(tx) = it.next() {
+                if tx.hash() == hash {
+                    return Some((tx, it.finish()));
+                }
+            }
+            None
+        };
+
+        let len = self.file.metadata()?.len();
+        self.file.seek(io::SeekFrom::Start(0))?;
+        if let Some((tx, CursorState { len, position, last_tx_len })) = search_transaction(hash, &self.file, len) {
```

Comment:
it might be nicer to exit early

```rust
let (tx, CursorState { len, position, last_tx_len}) = search_transaction(hash, &self.file, len) {
  Some(some) => some,
  None => return Ok(None);
}
```